File: internal/core/domain/response.go
package domain

type SuccessResponse struct {
	Message string      `json:"message"`
	Data    interface{} `json:"data,omitempty"`
}

type ErrorResponse struct {
	Message string `json:"message"`
	Errors  error  `json:"errors"`
}

type ValidationError struct {
	Field  string `json:"field"`
	Errors string `json:"errors,omitempty"`
}

File: frontend/src/lib/schema/response.ts
import { z } from "zod";

export const successSchema = z.object({
  message: z.string(),
  data: z.any().optional(),
});

export const errorSchema = z.object({
  message: z.string(),
});

export const errorValidationSchema = z.object({
  message: z.string(),
  errors: z.array(
    z.object({
      field: z.string(),
      error: z.string(),
    })
  ),
});

export type Success = z.infer<typeof successSchema>;
export type Error = z.infer<typeof errorSchema>;
export type ErrorValidation = z.infer<typeof errorValidationSchema>;
